ReactiveCocoaTests/Swift/SignalProducerSpec.swift

```diff
@@ -154,83 +274,427 @@ class SignalProducerSpec: QuickSpec {
+			it("should not trigger effects if disposed before closure return") {
+				var started = false
+				var value: Int?
+
+				SignalProducer<Int, NoError>(value: 42)
+					|> on(started: {
+						started = true
+					}, next: {
+						value = $0
+					})
+					|> startWithSignal { _, disposable in
+						expect(started).to(beFalsy())
+						expect(value).to(beNil())
+
+						disposable.dispose()
+					}
+
+				expect(started).to(beFalsy())
+				expect(value).to(beNil())
 			}
 
-			pending("should send interrupted if disposed before closure return") {
+			it("should send interrupted if disposed before closure return") {
+				var interrupted = false
+
+				SignalProducer<Int, NoError>(value: 42)
+					|> startWithSignal { signal, disposable in
+						expect(interrupted).to(beFalsy())
+
+						signal.observe(interrupted: {
+							interrupted = true
+						})
+
+						disposable.dispose()
+					}
+
+				expect(interrupted).to(beTruthy())
 			}
 
-			pending("should dispose of added disposables upon completion") {
+			it("should dispose of added disposables upon completion") {
+				let addedDisposable = SimpleDisposable()
+				var test: () -> () = { }
+
+				let producer = SignalProducer<Int, TestError>() { observer, disposable in
+					disposable.addDisposable(addedDisposable)
+					test = { _ in sendCompleted(observer) }
+				}
+
+				producer |> startWithSignal { _ in }
+				expect(addedDisposable.disposed).to(beFalsy())
+
+				test()
+				expect(addedDisposable.disposed).to(beTruthy())
 			}
 
-			pending("should dispose of added disposables upon error") {
+			it("should dispose of added disposables upon error") {
+				let addedDisposable = SimpleDisposable()
+				var test: () -> () = { }
+
+				let producer = SignalProducer<Int, TestError>() { observer, disposable in
+					disposable.addDisposable(addedDisposable)
+					test = { _ in sendError(observer, .Default) }
+				}
+
+				producer |> startWithSignal { _ in }
+				expect(addedDisposable.disposed).to(beFalsy())
+
+				test()
+				expect(addedDisposable.disposed).to(beTruthy())
 			}
 		}
 
 		describe("start") {
-			pending("should immediately begin sending events") {
+			it("should immediately begin sending events") {
+				let producer = SignalProducer<Int, NoError>(values: [1, 2])
+
+				var values: [Int] = []
+				var completed = false
+				producer.start(next: {
+					values.append($0)
+				}, completed: {
+					completed = true
+				})
+
+				expect(values).to(equal([1, 2]))
+				expect(completed).to(beTruthy())
 			}
 
-			pending("should send interrupted if disposed") {
+			it("should send interrupted if disposed") {
+				let producer = SignalProducer<(), NoError>.never
+
+				var interrupted = false
+				let disposable = producer.start(interrupted: {
+					interrupted = true
+				})
+
+				expect(interrupted).to(beFalsy())
+
+				disposable.dispose()
+				expect(interrupted).to(beTruthy())
 			}
 
 			pending("should release sink when disposed") {
```

Comment:
Can you add a `// FIXME` comment to this test, near the point of failure, so it's easy to find and remember what's wrong later on?